lib/resque/worker.rb

```diff
@@ -503,17 +503,15 @@ def heartbeat!(time = Time.now)
     end
 
     def start_heartbeat
-      heartbeat!
```

Comment:
I think this should be done outside the thread so that it is done synchronously resque starts processing jobs.  Otherwise, there is a race condition where the first job could start without a up to date heartbeat value written to redis. Unfortunately, race conditions aren't easy to write a failing test for.